src/helpers/augmentedReality/multipleSceneIndexGenerator.ts

```diff
@@ -24,36 +42,73 @@ export const multipleSceneIndexGenerator = ({
   let modelIndex = 0;
   const scenesCount = scenes?.length;
   let localUris = scenes?.[modelIndex]?.localUris;
+  const models: TModels = [];
+  const textures: TTextures = [];
 
   // if we have multiple scenes, we want to calculate the model and texture based on the current day
   // compared to the `startDate`
   if (scenesCount > 1 && startDate && timePeriodInDays) {
     // all models must have the same number of variable textures. therefore, in order to obtain the
     // number of textures, the textures of the first model were calculated.
-    const variableTextures = scenes[0]?.localUris?.filter(
+    let variableTextures = scenes[0]?.localUris?.filter(
       ({ type, stable }) => type === 'texture' && !stable
     );
     const variableTexturesCount = variableTextures?.length;
 
-    if (!variableTexturesCount) {
-      // nothing to do if there are no unstable textures
-      // keep original localUris with modelIndex = 0
-      return { localUris };
-    }
-
     const today = new Date();
     // the current day number is the number of running days since the given start date
     const currentDayNumber = extendedMoment.range(startDate, today).diff('days');
     const texture = Math.floor(currentDayNumber / timePeriodInDays);
     const model = Math.floor(texture / variableTexturesCount);
     modelIndex = model % scenesCount;
-    const textureIndex = texture % variableTexturesCount;
-    const variableTexture = variableTextures?.[textureIndex];
+
+    // When VariableTextures are created, they are created according to index 0 of the scene array.
+    // Once the modelIndex is found, it must be updated to select the correct texture file.
+    variableTextures = scenes[modelIndex]?.localUris?.filter(
+      ({ type, stable }) => type === 'texture' && !stable
+    );
     localUris = scenes?.[modelIndex]?.localUris;
+    const textureIndex = texture % variableTexturesCount;
+    const variableTexture = variableTextures[textureIndex];
```

Comment:
what if there is something wrong with `modelIndex` and no `variableTextures`?
`const variableTexture = variableTextures[textureIndex];` could crash.